foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -0,0 +1,277 @@
+            InputLocation endArtifactId = PomHelper.incrementColumn(startArtifactId, OLD_ARTIFACT_ID.length());
+
+            modifyPlugin.add(new PomModifications.Replacement(startGroupId, endGroupId, NEW_GROUP_ID));
+            modifyPlugin.add(new PomModifications.Replacement(startArtifactId, endArtifactId, NEW_ARTIFACT_ID));
+
+            writeModifications(file, modifyPlugin.finalizeMods());
+
+            Model updatedModel = getLocationAnnotatedModel(file);
+            Plugin updatedPlugin = getPlugin(updatedModel, NEW_ARTIFACT_ID);
+
+            if (IMAGE_VERSION != null && IMAGE_NAME != null) {
+                PomModifications modifyImages = new PomModifications();
+                // add new configuration code
+                int indent = 4;
+                int startConfigCodeRow = updatedPlugin.getLocation("artifactId").getLineNumber();
+                InputLocation startConfigCode = new InputLocation(startConfigCodeRow + 1, updatedPlugin.getLocation("artifactId").getColumnNumber());
+                modifyImages.add(new PomModifications.Insertion(startConfigCode, indent, OrphedomosToFabric8Migration::getConfigCode));
+
+                replacedConfig = writeModifications(file, modifyImages.finalizeMods());
+
+            }
+
+            Model fabric8Model = getLocationAnnotatedModel(file);
+            Plugin fabric8Plugin = getPlugin(fabric8Model, NEW_ARTIFACT_ID);
+            PomModifications removeOldConfig = new PomModifications();
+
+            // delete single line version tag if it exists
+            String version = fabric8Plugin.getVersion();
+            if (version != null){
+                InputLocation startVersionLocation = fabric8Plugin.getLocation("version" + LocationAwareMavenReader.START);
+                InputLocation endVersionLocation = fabric8Plugin.getLocation("version" + LocationAwareMavenReader.END);
+
+                PomModifications.Deletion pomDeletePreviousVersion = new PomModifications.Deletion(startVersionLocation, endVersionLocation);
+                removeOldConfig.add(pomDeletePreviousVersion);
+            }
+
+            // delete old orphedomos configuration
+            InputLocation startConfigLocation = fabric8Plugin.getLocation("configuration" + LocationAwareMavenReader.START);
+            InputLocation endConfigLocation = fabric8Plugin.getLocation("configuration" + LocationAwareMavenReader.END);
+            PomModifications.Deletion pomDeletePreviousConfig = new PomModifications.Deletion(startConfigLocation, endConfigLocation);
+            removeOldConfig.add(pomDeletePreviousConfig);
+
+            writeModifications(file, removeOldConfig.finalizeMods());
+        }
+
+        // check if pom file has <packaging>orphedomos</packaging> and replace
+        if (containsOrphedomosPackaging(file)){
+            replacedPackaging = executePackagingMigration(file);
+        }
+
+        isSuccessful = replacedConfig || replacedPackaging;
+
+        return isSuccessful;
```

Comment:
A: this isn't a must, but this function just feels big. Might be nice to break this up into a few subroutines